checkout-core/src/main/java/com/adyen/checkout/core/internal/data/api/HttpClientExt.kt

```diff
@@ -90,10 +87,19 @@ private fun HttpException.getLogMessage(): String {
     }
 }
 
-private fun ByteArray.toJSONObject(): JSONObject {
+private fun String.toJSONObject(): JSONObject {
     return if (isEmpty()) {
         JSONObject()
     } else {
-        JSONObject(String(this, Charsets.UTF_8))
+        JSONObject(this)
     }
 }
+
+private fun Any.logResponse(response: AdyenApiResponse) {
+    adyenLog(AdyenLogLevel.VERBOSE) { "response - ${response.statusCode} .../${response.path}" }
+    response.headers.forEach { (key, value) ->
+        adyenLog(AdyenLogLevel.VERBOSE) { "$key: $value" }
+    }
+    adyenLog(AdyenLogLevel.VERBOSE) { response.body }
```

Comment:
Could we use `response.body.toJSONObject().toStringPretty()` to make the body look nicer?